numpy/core/src/multiarray/shape.c

```diff
@@ -680,9 +680,9 @@ PyArray_SwapAxes(PyArrayObject *ap, int a1, int a2)
 NPY_NO_EXPORT PyObject *
 PyArray_Transpose(PyArrayObject *ap, PyArray_Dims *permute)
```

Comment:
Looking at `c-api.array.rst`, ever occurence of `axis` is an `int`, not an `intp`. So from a consistency point of view, passing `int` would be better. Also, this bug wouldn't have existed if this used int in the first place like everything else.

`int` is not strictly necessary, but it is advisable. Of course, changing the public API here is a lot of work for little gain.